src/main/groovy/de/itemis/mps/gradle/CreateDmg.groovy

```diff
@@ -59,11 +73,19 @@ class CreateDmg extends DefaultTask {
                             'Mac/Finder/DSStore/BuddyAllocator.pm', 'Mac/Finder/DSStore.pm']
         File scriptsDir = File.createTempDir()
         File dmgDir = File.createTempDir()
+        def getSigningInfo = [signKeyChainPassword, signKeyChain, signIdentity]
         try {
             BundledScripts.extractScriptsToDir(scriptsDir, scripts)
             project.exec {
                 executable new File(scriptsDir, 'mpssign.sh')
-                args rcpArtifact, dmgDir, jdk
+
+                if(getSigningInfo.every {el -> el != null}) {
+                    args '-r', rcpArtifact, '-o', dmgDir, '-j', jdk, '-p', signKeyChainPassword, '-k', signKeyChain, '-i', signIdentity
+                }else if (getSigningInfo.every {el -> el == null}){
```

Comment:
`{ el -> el == null }` can be shortened to `{ it == null }` since `it` is the implicit closure parameter in Groovy (http://groovy-lang.org/closures.html#implicit-it)

Likewise `{ el -> el != null }` two lines above this one.